src/EnergyPlus/DataSystemVariables.cc

```diff
@@ -255,81 +244,44 @@ namespace DataSystemVariables {
-            FileFound = true;
-            CheckedFileName = DataStringGlobals::inputDirPathName + InputFileName;
-            print(ioFiles.audit, "{}={}\n", "found (input file)", getAbsolutePath(CheckedFileName));
-            return;
-        } else {
-            print(ioFiles.audit, "{}={}\n", "not found (input file)", getAbsolutePath(DataStringGlobals::inputDirPathName + InputFileName));
-        }
-
-        // Look relative to input path
-        if (FileSystem::fileExists(envinputpath1 + InputFileName)) {
-            FileFound = true;
-            CheckedFileName = envinputpath1 + InputFileName;
-            print(ioFiles.audit, "{}={}\n", "found (epin)", getAbsolutePath(CheckedFileName));
-            return;
-        } else {
-            print(ioFiles.audit, "{}={}\n", "not found (epin)", getAbsolutePath(envinputpath1 + InputFileName));
-        }
-
-        // Look relative to input path
-        if (FileSystem::fileExists(envinputpath2 + InputFileName)) {
-            FileFound = true;
-            CheckedFileName = envinputpath2 + InputFileName;
-            print(ioFiles.audit, "{}={}\n", "found (input_path)", getAbsolutePath(CheckedFileName));
-            return;
-        } else {
-            print(ioFiles.audit, "{}={}\n", "not found (input_path)", getAbsolutePath(envinputpath2 + InputFileName));
-        }
-
-        // Look relative to program path
-        if (FileSystem::fileExists(envprogrampath + InputFileName)) {
-            FileFound = true;
-            CheckedFileName = envprogrampath + InputFileName;
-            print(ioFiles.audit, "{}={}\n", "found (program_path)", getAbsolutePath(CheckedFileName));
-            return;
-        } else {
-            print(ioFiles.audit, "{}={}\n", "not found (program_path)", getAbsolutePath(envprogrampath + InputFileName));
-        }
-
-        if (!TestAllPaths) return;
-
-        // Look relative to current working folder
-        if (FileSystem::fileExists(CurrentWorkingFolder + InputFileName)) {
-            FileFound = true;
-            CheckedFileName = CurrentWorkingFolder + InputFileName;
-            print(ioFiles.audit, "{}={}\n", "found (CWF)", getAbsolutePath(CheckedFileName));
-            return;
-        } else {
-            print(ioFiles.audit, "{}={}\n", "not found (CWF)", getAbsolutePath(CurrentWorkingFolder + InputFileName));
+        std::vector<std::pair<std::string, std::string>> pathsChecked;
+
+        const std::vector<std::pair<std::string, std::string>> pathsToCheck = {
+            {InputFileName, "InputFileName"},
+            {DataStringGlobals::inputDirPathName + InputFileName, "inputDirPathName"},
+            {envinputpath1 + InputFileName, "envinputpath1"},
+            {envinputpath2 + InputFileName, "envinputpath2"},
+            {envprogrampath + InputFileName, "envprogrampath"},
+            {CurrentWorkingFolder + InputFileName, "CurrentWorkingFolder"},
+            {ProgramPath + InputFileName, "ProgramPath"}
```

Comment:
@ajscimone Here are more informative descriptions for each of the strings that describe where files are searched:

1. Current working directory
2. IDF directory
3. "epin" environment variable
4. "input_path" environment variable
5. <This variable is never used or set, I think it's supposed to be "program_path" environment variable initialized as `cProgramPath`, but that variable overwrites `ProgramPath` instead>
6. INI directory (if it exists) <from what I can tell this is always identical to [1] -- maybe we can drop it?>
7. "program" > "dir" from INI file

We should maybe add `exeDirectory` to the paths searched, too? I think that is the intent of [5] and [7], but neither of those seem reliable.

@Myoldmopar @mjwitte thoughts on any of this?

Separate question: Does/should this interact with the way python plug-in paths are searched?